pkg/appmanager/resourceConfig.go

```diff
@@ -1498,10 +1848,104 @@ func (appMgr *Manager) handleRouteTls(
 				rc.Virtual.AddIRule(passThroughIRuleName)
 				if !abDeployment {
 					rc.AddRuleToPolicy(policyName, rule)
+					setAnnotationRulesForRoute(policyName, virtualName, poolName, urlRewriteRule, appRootRules, rc, appMgr)
+				}
+			}
+		}
+	}
+	if urlRewriteRule != nil || len(appRootRules) != 0 {
+		rc.MergeRules(appMgr.mergedRulesMap)
+	}
+}
+
+func (appMgr *Manager) AddRuleToAnnotationsMap(virtualName, poolName, ruleName string) {
+	if _, ok := appMgr.annotationRulesMap[virtualName]; !ok {
+		appMgr.annotationRulesMap[virtualName] = make(map[string][]string)
+	}
+	appMgr.annotationRulesMap[virtualName][poolName] = append(appMgr.annotationRulesMap[virtualName][poolName], ruleName)
```

Comment:
You should check if the rule you are about to add is already in the list. If it is, we don't want to add it again.

Reply:
Good catch I'll fix that